app/src/main/java/org/coepi/android/ui/notifications/NotificationShower.kt

```diff
@@ -35,12 +32,19 @@ class NotificationsShower(
         }
     }
 
-    private fun pendingIntent(args: NotificationIntentArgs): PendingIntent = PendingIntent.getActivity(
-        context, 0, Intent(context, MainActivity::class.java).apply {
-            flags = FLAG_ACTIVITY_SINGLE_TOP
-            putExtra(args.key.toString(), args.value)
-        }, FLAG_UPDATE_CURRENT
-    )
+    fun cancelNotification() {
+        val notificationManager: NotificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        notificationManager.cancel(LocalNotificationChannelId.INFECTION_REPORTS_CHANNEL.ordinal)
```

Comment:
Since this class is generic, the notification to be cancelled should be passed as a parameter.

Also, passing `LocalNotificationChannelId` here seems incorrect and it's working probably by coincidence. This expects a notification id.